### DISCUSSION TITLE
Generate new passwords.yaml if absent

### DISCUSSION THREAD

**Discussion by andriyndev** [serverpod/serverpod#863] [Ideas]
I think Serverpod should generate new `passwords.yaml` file if it's absent. It might happen if the project is clonned from GitHub where it's pushed without `passwords.yaml` because the latest contains sensitive information. In this case we get the "Failed to load passwords. Serverpod cannot not start." error when trying to start the server.
Upd. And fix the typo too: "cannot not" -> "cannot".

**Comment by vlidholt** [comment]
Even better, there should be clear output when Serverpod starts that the file is missing.

**Reply by andriyndev** [reply]
I agree, but there should also be an easy way to regenerate the default file otherwise the developer will have to search for one that is not very convenient. As a solution - add the corresponding command to the Serverpod command line tool.

**Reply by tgrushka** [reply]
While discussing missing startup files, while this is probably a separate issue, I had to dig for my setup-tables script (and wound up customizing it anyway), because it gets deleted after setup.

Once you initialize a serverpod project, this script gets deleted, so if you run `docker compose down --volumes` because you want to clean the slate and start over, you can't regenerate the tables without scripting it or finding the setup-tables script in the serverpod source. Would be nice if that could be included in this same theme as regenerating the passwords file.

Here is my customized `setup-tables`:

```
#!/bin/sh

set -e

# brew install yq
DB_PASSWORD=$(yq '.development.database' < config/passwords.yaml)
PROJECT=intone

# Clean docker volumes
docker compose down --volumes

# Start docker
echo "Starting docker"
docker compose up --build --detach

if command -v netstat &> /dev/null
then
  while ! netstat -tna | grep 'LISTEN' | grep -q '.8090'; do
    echo "Waiting for Postgres..."
    sleep 2 # time in seconds, tune it as needed
  done
else
  while ! ss -tulpn | grep LISTEN | grep :8090; do
    echo "Waiting for Postgres..."
    sleep 2
  done
fi

sleep 2

echo "Postgres is ready"

cat generated/*.pgsql | docker compose run -T --rm postgres env PGPASSWORD="$DB_PASSWORD" psql -h postgres -U postgres -d $PROJECT -v ON_ERROR_STOP=1

echo "Restarting docker"
docker compose stop
docker compose up
```

**Reply by vlidholt** [reply]
We are adding code that will automatically setup any missing tables (and handle migrations) in version 1.2, so this will no longer be an issue then. :)
